lib/declare_schema/model/habtm_model_shim.rb

```diff
@@ -53,8 +57,7 @@ def _declared_primary_key
 
       def index_definitions_with_primary_key
         [
-          IndexDefinition.new(self, foreign_keys, unique: true, name: ::DeclareSchema::Model::IndexDefinition::PRIMARY_KEY_NAME),
-          IndexDefinition.new(self, foreign_keys.last) # not unique by itself; combines with primary key to be unique
```

Comment:
I think this was intentionally added in order to keep the `change_index` method in parity with what the actual active_record indexes look like. I added a workaround in fb7f371f9876dcdbb49e8925ebd9b6bd3ad07a51 but maybe we want to add this back and be explicit about this definition.

Reply:
Looking at the other fix you put in, this is much more explicit and can be more easily understood what is happening here. For that reason I think we should leave the old implementation over the more obscure `.pop` that was added in the linked commit.